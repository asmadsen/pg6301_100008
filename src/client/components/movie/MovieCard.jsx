import React from 'react'
import PropTypes from 'prop-types'
import CardContent from '@material-ui/core/CardContent'
import { makeStyles, Typography } from '@material-ui/core'
import MovieGenres from './MovieGenres'
import CardActions from '@material-ui/core/CardActions'
import Button from '@material-ui/core/Button'
import { AdapterLink } from '../../utils/Router'
import Card from '@material-ui/core/Card'

const useStyles = makeStyles({
	spacing: {
		flexGrow: 1
	},
	cardHeader: {
		display: 'flex',
		justifyContent: 'space-between'
	}
})

const MovieCard = ({ movie }) => {
	const classes = useStyles()

	return (
		<Card>
			<CardContent>
				<div className={classes.cardHeader}>
					<Typography component="h5" variant="h5">{movie.title}</Typography>
					<Typography component="span" variant="subtitle1">{movie.rating.toFixed(1)}</Typography>
				</div>
				<MovieGenres genres={movie.genres.slice(0, 3)} />
				<Typography variant="body2" component="p">
					{movie.plot}
				</Typography>
			</CardContent>
			<CardActions>
				<div className={classes.spacing}/>
				<Button
					size="small"
					component={AdapterLink}
					to={`/movies/${movie.id}`}
				>
					Read More
				</Button>
			</CardActions>
		</Card>
	)
}

MovieCard.propTypes = {
	movie: PropTypes.shape({
		id: PropTypes.number,
		title: PropTypes.string,
		genres: PropTypes.arrayOf(PropTypes.string),
		plot: PropTypes.string,
		rating: PropTypes.number
	})
}

export default MovieCard
