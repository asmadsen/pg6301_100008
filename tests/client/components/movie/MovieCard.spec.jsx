import MovieCard from '../../../../src/client/components/movie/MovieCard'
import React from 'react'
import { shallow } from 'enzyme'

describe('client/component/movie/MovieCard', () => {
	it('should match snapshot', () => {
		const movie = {
			id: 1,
			title: 'Die Hard',
			genres: ['Action', 'Drama'],
			plot: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
			rating: 3.6
		}
		const wrapper = shallow(<MovieCard movie={movie} />)

		expect(wrapper).toMatchSnapshot()
	})
})
