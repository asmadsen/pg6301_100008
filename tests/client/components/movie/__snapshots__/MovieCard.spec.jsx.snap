// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client/component/movie/MovieCard should match snapshot 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardContent))>
    <div
      className="makeStyles-cardHeader-2"
    >
      <WithStyles(ForwardRef(Typography))
        component="h5"
        variant="h5"
      >
        Die Hard
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        component="span"
        variant="subtitle1"
      >
        3.6
      </WithStyles(ForwardRef(Typography))>
    </div>
    <MovieGenres
      genres={
        Array [
          "Action",
          "Drama",
        ]
      }
    />
    <WithStyles(ForwardRef(Typography))
      component="p"
      variant="body2"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit.
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(CardActions))>
    <div
      className="makeStyles-spacing-1"
    />
    <WithStyles(ForwardRef(Button))
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      size="small"
      to="/movies/1"
    >
      Read More
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
